packages/mrml-core/src/mj_carousel/render.rs

```diff
@@ -24,8 +26,26 @@ struct MjCarouselExtra {
 }
 
 impl<'root> Renderer<'root, MjCarousel, MjCarouselExtra> {
+    #[cfg(feature = "fragment")]
+
+    fn children_iter(&self) -> impl Iterator<Item = &MjCarouselChild> {
+        fn folder<'root>(
+            c: &'root MjCarouselChild,
+        ) -> Box<dyn Iterator<Item = &MjCarouselChild> + 'root> {
+            match c {
+                MjCarouselChild::Fragment(f) => Box::new(f.children.iter().flat_map(folder)),
+                _ => Box::new(std::iter::once(c)),
+            }
+        }
+        self.element.children.iter().flat_map(folder)
+    }
+
+    #[cfg(not(feature = "fragment"))]
+    fn children_iter(&self) -> impl Iterator<Item = &MjCarouselChild> {
+        self.element.children.iter()
+    }
     fn get_thumbnails_width(&self) -> Pixel {
-        let count = self.element.children.len();
+        let count = self.children_iter().count();
```

Comment:
having a `children_count()` function would allow, when the fragment feature is disabled, to not go through the whole iterator just to get the count